feat: add `utils/none-in-by`

AgPriyanshu18 opened the pull request:
Check whether every property of object does not satisfy predicate function

Fixes #824

Resolves #824 .

## Description

> What is the purpose of this pull request?

This pull request:

-   a add utils/none-in-by
-   b Check whether every property of object does not satisfy predicate function
-   c

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #824 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

AgPriyanshu18 committed:
feat: added utils/none-in-by

Check whether every property of object does not satisfy predicate function

Fixes #824

Jaysukh-409 commented on lib/node_modules/@stdlib/utils/none-in-by/benchmark/julia/benchmark.jl:
```diff
@@ -0,0 +1,145 @@
+#
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

Jaysukh-409 reviewed (commented):
Change Copyright years wherever it is necessary.

Jaysukh-409 commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,134 @@
+
+	t.strictEqual(bool, false, 'returns false');
+	t.end();
+});
```
Needs to add test for providing an execution context.

Jaysukh-409 commented on lib/node_modules/@stdlib/utils/none-in-by/README.md:
````diff
@@ -0,0 +1,165 @@
+var bool = noneInBy( obj, isUnderage );
+// returns false
+```
+
````
Add example of the function execution context and what predicate function receives as an argument.

AgPriyanshu18 committed:
docs: added example of function execution context and related tests

applied all suggested changes

Fixes #824

AgPriyanshu18 committed:
fix: fix the function execution context tests

fixed the bug in function execution test

Fixes #824

AgPriyanshu18 committed:
fix: fix the function execution context tests

fixed the bug in function execution test

Fixes #824

AgPriyanshu18 commented:
@Jaysukh-409 I have made all suggested changes.

Pranavchiku commented on lib/node_modules/@stdlib/utils/none-in-by/README.md:
```diff
@@ -0,0 +1,202 @@
+
+# noneInBy
+
+> Tests whether every property of an object fails a test implemented by a predicate function.
```
```suggestion
> Test whether every property of an object fails a test implemented by a predicate function.
```

Pranavchiku commented on lib/node_modules/@stdlib/utils/none-in-by/README.md:
````diff
@@ -0,0 +1,202 @@
+
+-   If provided an empty object, the function returns `true`.
+
+```javascript
+function truthy() {
+    return true;
+}
+var bool = noneInBy( {}, truthy );
+// returns true
+```
````
```suggestion
    ```javascript
    function truthy() {
        return true;
    }
    var bool = noneInBy( {}, truthy );
    // returns true
    ```
```

Pranavchiku commented on lib/node_modules/@stdlib/utils/none-in-by/README.md:
````diff
@@ -0,0 +1,202 @@
+var bool = noneInBy( {}, truthy );
+// returns true
+```
+
````
mIssing ``The function does **not** skip `undefined` elements.``

Pranavchiku reviewed (changes requested):
This PR needs a few changes, I have requested changes and ask you to iterate over entire package comparing against `stdlib/utils/none-by`. Thank you!

AgPriyanshu18 commented:
@Pranavchiku I have completed the suggested changes and have reviewed the whole package.

AgPriyanshu18 committed:
docs: fix style and added missing function in readme

Fixed the style in readme and added code block for function does not skip ndefines elements

Fixes #824

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/docs/repl.txt:
```diff
@@ -0,0 +1,42 @@
+
+{{alias}}( object, predicate[, thisArg ] )
+    Tests whether every property in a collection fail a test implemented by a
```
```suggestion
    Tests whether every property in an object fails a test implemented by a
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/docs/types/test.ts:
```diff
@@ -0,0 +1,61 @@
+	'a':1,
+	'b':2,
+	'c':4
+};
```
```suggestion
const obj = {
	'a': 1,
	'b': 2,
	'c': 4
};
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/lib/index.js:
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+/**
+* Test whether every property of a object fail a test implemented by a predicate function.
```
```suggestion
* Test whether every property of an object fails a test implemented by a predicate function.
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/lib/index.js:
```diff
@@ -0,0 +1,46 @@
+*     return ( v > 0 );
+* }
+*
+* var obj = { 'a' : -1, 'b' : -2, 'c' : -3, 'd' : -4};
```
```suggestion
* var obj = { 'a' : -1, 'b' : -2, 'c' : -3, 'd' : -4 };
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/lib/main.js:
```diff
@@ -0,0 +1,71 @@
+* @param {Object} obj - input object
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @throws {TypeError} first argument must be a object
```
```suggestion
* @throws {TypeError} first argument must be an object
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/lib/main.js:
```diff
@@ -0,0 +1,71 @@
+	var key;
+
+	if ( !isObject( obj ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a object. Value: `%s`.', obj ) );
```
```suggestion
		throw new TypeError( format( 'invalid argument. First argument must be an object. Value: `%s`.', obj ) );
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/package.json:
```diff
@@ -0,0 +1,71 @@
+{
+    "name": "@stdlib/utils/none-in-by",
+    "version": "0.0.0",
+    "description": "Test whether every property in a collection fail a test implemented by a predicate function.",
```
```suggestion
    "description": "Test whether every property in an object fails a test implemented by a predicate function.",
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/package.json:
```diff
@@ -0,0 +1,71 @@
+      "array.every",
+      "iterate",
+      "collection",
+      "array-like",
```
```suggestion
      "every",
      "iterate",
      "object",
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,167 @@
+	t.end();
+});
+
+tape('the function returns `true` if every property fail a test ', function test(t) {
```
```suggestion
tape('the function returns `true` if every property fails a test ', function test(t) {
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,167 @@
+	t.end();
+});
+
+tape('the function throws an error if not provided an object', function test(t) {
```
```suggestion
tape( 'the function throws an error if not provided an object', function test( t ) {
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,167 @@
+	}
+});
+
+tape('if provided an empty object, the function returns `true`', function test(t) {
```
```suggestion
tape( 'if provided an empty object, the function returns `true`', function test( t ) {
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,167 @@
+	t.end();
+});
+
+tape('the function returns `false` if one or more properties pass a test ', function test(t) {
```
```suggestion
tape( 'the function returns `false` if one or more properties pass a test ', function test( t ) {
```

Planeshifter commented on lib/node_modules/@stdlib/utils/none-in-by/test/test.js:
```diff
@@ -0,0 +1,167 @@
+	t.strictEqual(ctx.sum / ctx.count, 2.0, 'expected result');
+
+	t.end();
+});
```
Should update spacing around parentheses per our code style:

```suggestion
tape( 'the function supports providing an execution context', function test( t ) {
	var bool;
	var ctx;
	var obj;

	function sum( value ) {
		/* eslint-disable no-invalid-this */
		if ( value < 0 ) {
			return false;
		}
		this.sum += value;
		this.count += 1;
		return false;
	}

	ctx = {
		'sum': 0,
		'count': 0
	};
	obj = {
		'0': 1.0,
		'1': 2.0,
		'2': 3.0
	};

	bool = noneInBy( obj, sum, ctx );

	t.strictEqual( bool, true, 'returns true' );
	t.strictEqual( ctx.sum / ctx.count, 2.0, 'expected result' );

	t.end();
});
```

Planeshifter reviewed (approved):
Looks good; just made a few code style and documentation fixes, but otherwise this PR should be ready to land once CI clears.

Planeshifter committed:
Apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Update lib/node_modules/@stdlib/utils/none-in-by/test/test.js

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>
